BPR.Mediator/Services/ResultService.cs

```diff
@@ -0,0 +1,114 @@
+using AutoMapper;
+using BPR.Analysis.Mappers;
+using BPR.Analysis.Models;
+using BPR.Analysis.Services;
+using BPR.Mediator.Enums;
+using BPR.Mediator.Models;
+using BPR.Mediator.Validators;
+using BPR.Persistence.Models;
+using BPR.Persistence.Repositories;
+using BPR.Persistence.Utils;
+using Microsoft.Extensions.Logging;
+
+namespace BPR.Mediator.Services;
+
+public class ResultService : IResultService
+{
+    private readonly IResultRepository _resultRepository;
+    private readonly IAnalysisService _analysisService;
+    private readonly IValidatorService _validatorService;
+    private readonly IMapper _mapper;
+    private readonly ILogger<ResultService> _logger;
+    
+    public ResultService(IResultRepository resultRepository, IAnalysisService analysisService, IValidatorService validatorService, IMapper mapper, ILogger<ResultService> logger)
+    {
+        _analysisService = analysisService;
+        _resultRepository = resultRepository;
+        _validatorService = validatorService;
+        _mapper = mapper;
+        _logger = logger;
+    }
+    
+    public async Task<IList<ResultModel>> GetAllResultsAsync()
+    {
+        var result = (await _resultRepository.GetAllResultsAsync()).Val;
```

Comment:
is the Val from a wrapper class? if it is ours maybe we should rename it to Value